app/services/claimToAdjustPoa/SubmitPoaHelper.scala

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services.claimToAdjustPoa
+
+import auth.MtdItUser
+import config.featureswitch.FeatureSwitching
+import config.{AgentItvcErrorHandler, ItvcErrorHandler}
+import controllers.agent.predicates.ClientConfirmedController
+import controllers.routes.HomeController
+import models.admin.AdjustPaymentsOnAccount
+import models.claimToAdjustPoa.{PaymentOnAccountViewModel, PoAAmendmentData}
+import models.core.Nino
+import play.api.Logger
+import play.api.mvc.Result
+import services.{ClaimToAdjustPoaCalculationService, ClaimToAdjustService, PaymentOnAccountSessionService}
+import uk.gov.hmrc.http.HeaderCarrier
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait SubmitPoaHelper extends ClientConfirmedController with FeatureSwitching {
+
+  protected def dataFromSession(poaSessionService: PaymentOnAccountSessionService)(implicit hc: HeaderCarrier, ec: ExecutionContext)
+  : Future[PoAAmendmentData] = {
+    poaSessionService.getMongo(hc, ec).flatMap {
+      case Right(Some(newPoaData: PoAAmendmentData)) =>
+        Future.successful(newPoaData)
+      case _ =>
+        Future.failed(new Exception(s"Failed to retrieve session data"))
+    }
+  }
+
+  protected def handlePoaAndOtherData(poa: PaymentOnAccountViewModel,
+                                      otherData: PoAAmendmentData, isAgent: Boolean, nino: String, ctaCalculationService: ClaimToAdjustPoaCalculationService)
```

Comment:
:seedling: - in general if we can use Nino value class instead of String type, but should be fine to use String this time;

Reply:
I was just refactoring the code but happy to add this